modules/ingester/ingester.go

```diff
@@ -148,6 +148,14 @@ func (i *Ingester) loop(ctx context.Context) error {
 func (i *Ingester) stopping(_ error) error {
 	i.markUnavailable()
 
+	if i.cfg.FlushAllOnShutdown {
+		i.lifecycler.SetUnregisterOnShutdown(true)
```

Comment:
i don't think we should set this. allow this value to be configured independently of `flush_all_on_shutdown`.

Reply:
But i see it being done in flush.ShutDownHandler. Shouldn't we have same behaviour for stopping too?

Reply:
So now ShutDownHandler too would flush only if this new flag is true?

Reply:
No. Shutdown should continue to do exactly what it does. It should not require a flag to set set.